tests/Benchmarks/Microsoft.Experimental.Collections/MultiValueDictionaryPerformanceTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using BenchmarkDotNet.Attributes;
+
+namespace Microsoft.Collections.Extensions.Tests
+{
+    [MemoryDiagnoser]
+    public class MultiValueDictionaryPerformanceTests
+    {
+        [Params(1000, 10_000, 100_000)]
+        public int Size { get; set; }
+
+        private int _randomKey;
+        private List<int> _values;
+        private MultiValueDictionary<int, int> _dict;
+        private int _value;
+
+        [GlobalSetup(Targets = new[] { nameof(Clear), nameof(GetKeys) })]
+        public void CreateMultiValueDictionary_SingleValues()
+        {
+            _dict = new MultiValueDictionary<int, int>();
+            Random rand = new Random(11231992);
+
+            while (_dict.Count < Size)
+                _dict.Add(rand.Next(), rand.Next());
```

Comment:
what if the dictionary contains given key? will `Add` throw an exception?

Reply:
No, since this is a multivalue dictionary it allows to have multiple values for the same key, so if the key already exists it just adds the value to the key's collection of values. 

https://github.com/dotnet/corefxlab/blob/b17275841c60ece2e5e8e8b2f089588f4ca91a2e/src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/MultiValueDictionary.cs#L637